cmd/metal-api/internal/service/network-service.go

```diff
@@ -512,6 +594,31 @@ func (r *networkResource) allocateNetwork(request *restful.Request, response *re
 		destPrefixes = append(destPrefixes, *prefix)
 	}
 
+	r.log.Info("network allocate", "partition", partition.ID)
+	var (
+		superNetwork metal.Network
+	)
+
+	err = r.ds.FindNetwork(&datastore.NetworkSearchQuery{
+		PartitionID:     &partition.ID,
+		PrivateSuper:    pointer.Pointer(true),
+		ParentNetworkID: requestPayload.ParentNetworkID,
+	}, &superNetwork)
+	if err != nil {
+		r.sendError(request, response, defaultError(err))
+		return
+	}
+
+	if superNetwork.ID == "" {
+		r.sendError(request, response, httperrors.BadRequest(fmt.Errorf("no supernetwork found")))
+		return
+	}
+	if len(superNetwork.DefaultChildPrefixLength) == 0 {
+		r.sendError(request, response, httperrors.BadRequest(fmt.Errorf("supernetwork %s has no defaultchildprefixlength specified", superNetwork.ID)))
```

Comment:
Should be an internal server error but I am not sure if we need these validation because we ensure this somewhere else in the code?